### ISSUE TITLE
wallet: #17261 regression: addmultisigaddress addresses not treated as solvable

### ISSUE THREAD

**Issue by ryanofsky** [bitcoin/bitcoin#18075]
Filing this issue to record steps to reproduce a small regression from #17261, which is fixed in #18067. Reproducing the bug has to be done manually because we don't have a framework for writing wallet backwards compatibility tests (#12134 is stuck in review)

# Steps to reproduce

```sh
MERGE=2d6e76af240969aa284cd4c3d376493988e218c2 # 17261 merge commit
PK1=0296b538e853519c726a2c91e61ec11600ae1390813a627c66fb8be7947be63c52 # key from wallet_address_types.py
PK2=037211a824f55b505228e4c3d5194c1fcfaa15a456abdf37f9b9d97a4040afc073 # key from wallet_address_types.py

git checkout $MERGE^
make
src/bitcoind -regtest &
ADDRESS=$(src/bitcoin-cli -regtest addmultisigaddress 1 "[\"$PK1\", \"$PK2\"]" true legacy | jq -r .address)
src/bitcoin-cli -regtest stop

git checkout $MERGE
make -C src bitcoind
src/bitcoind -regtest &
src/bitcoin-cli -regtest getaddressinfo $ADDRESS | jq .solvable # BUG: expect true, actual false
src/bitcoin-cli -regtest stop
```

# Expected behavior

getaddressinfo reports multisig address is solvable

# Actual behavior

getaddressinfo returns `"solvable": false`

**Comment by Sjors** [comment]
I wrote a test on top of #12134 that reproduces this by calling `addmultisigaddress` in v0.19.0.1, loads that wallet on master and fails. #18067 fixes it: https://github.com/Sjors/bitcoin/commit/bacf55990905c0257c7bc45bbbd2b028716efc7f

**Comment by Empact** [comment]
According to @achow101 this should be fixed now that #18067 is merged.
